pkg/services/ngalert/remote/client/mimir.go

```diff
@@ -60,21 +64,21 @@ func (e *errorResponse) Error() string {
 	return e.Error2
 }
 
-func New(cfg *Config) (*Mimir, error) {
+func New(cfg *Config, metrics *metrics.RemoteAlertmanager) (*Mimir, error) {
 	rt := &MimirAuthRoundTripper{
 		TenantID: cfg.TenantID,
 		Password: cfg.Password,
 		Next:     http.DefaultTransport,
 	}
-
-	c := http.Client{
+	c := &http.Client{
 		Transport: rt,
 	}
 
 	return &Mimir{
 		endpoint: cfg.URL,
-		client:   c,
+		client:   client.NewTimedClient(c, metrics.HTTPRequestsDuration),
```

Comment:
@alexweav put together a PR to remove the dependency on weaveworks common: https://github.com/grafana/grafana/pull/80051

We should merge his PR and rebase this one when ready.

Reply:
yeah this is merged, though I put `client.NewTimedClient` in a place that's probably hard to access from this package since it was only used in one spot.

https://github.com/grafana/grafana/blob/37532cc5fa8460f8ad0b73c7ec3fd1fd65b07e05/pkg/services/ngalert/state/historian/client.go

Might need to move it to a shared package? Or we could put it in alerting? Wherever is easiest for you to consume